### PR TITLE
[14.0][IMP] account_payment_order

### PR DESCRIPTION
**Pull request by max3903**

### CODE REVIEW

**Review by mileo — APPROVED** [review]

**Review by marcelsavegnago — APPROVED** [review]

**Review by HaraldPanten — APPROVED** [review]

File: account_payment_order/models/bank_payment_line.py [hunk 1]
```diff
@@ -150,42 +150,16 @@ def reconcile(self):
...
-            lines_to_rec += payment_line.move_line_id
-
-        lines_to_rec.reconcile()
+        for line in transit_mlines:
+            lines_to_rec = line
+            for payment_line in self.payment_line_ids.filtered(
+                lambda x: x.move_line_id
+                and not x.move_line_id.reconciled
+                and x.move_line_id.account_id == line.account_id
+                and x.move_line_id.credit == line.debit
```
Comment (alexis-via) [inline account_payment_order/models/bank_payment_line.py:159]: you cannot compare 2 floats with ==

File: account_payment_order/views/account_payment_order.xml [hunk 2]
```diff
@@ -40,6 +40,12 @@
                         states="cancel"
                         string="Back to Draft"
                     />
+                    <button
+                        name="action_done"
+                        type="object"
+                        states="uploaded"
+                        string="Done Payments"
+                    />
```
Comment (alexis-via) [inline account_payment_order/views/account_payment_order.xml:48]: What's the point of the "done" state ? It was not present in my initial design of account_payment_order, and I see that it has been introduced by @JordiBForgeFlow in this commit
https://github.com/OCA/bank-payment/commit/81152bf60304b9fde3629e038056c096f956ebf5
but the commit message gives no clue about the reason for this new "done" state. For me, the final state is "uploaded"... what's the added value of the done state?
Sorry if my question is stupid...

File: account_payment_order/models/account_payment_order.py [hunk 3]
```diff
@@ -462,7 +462,7 @@ def _prepare_move_line_offsetting_account(
             vals.update({"date_maturity": bank_lines[0].date})
 
         if self.payment_mode_id.offsetting_account == "bank_account":
-            account_id = self.journal_id.default_account_id.id
+            account_id = self.journal_id.payment_debit_account_id.id
```
Comment (alexis-via) [inline account_payment_order/models/account_payment_order.py:469]: you should make a difference between inbound and outbond : payment_debit_account_id and payment_credit_account_id

**Comment by dreispt** [comment]
Travis is red.

File: account_payment_order/views/account_payment_order.xml (hunk 2 above)
Comment (hbrunn) [reply account_payment_order/views/account_payment_order.xml:48]: I'm attempting to fix this soon along the lines of what previous versions did, see https://github.com/OCA/bank-payment/pull/784#issuecomment-896785175 - when done we can add this here too in another PR

**Comment by atchuthan** [comment]
Cannot use git-aggreagate, please resolve conflicts
Is there any simple alternative or should I add the fixes manually for our instance?

@SodexisTeam

**Comment by dreispt** [comment]
...I rebased and solved conflicts.

**Comment by atchuthan** [comment]
Thanks @dreispt

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Pending comments inline

**Comment by pedrobaeza** [comment]
And please provide a description of the "improvements".

File: account_payment_order/views/account_payment_order.xml (hunk 2 above)
Comment (atchuthan) [reply account_payment_order/views/account_payment_order.xml:48]:
```
2021-11-09 14:34:02,634 28 INFO KEN_Master_V14_2021_09_23 odoo.addons.base.models.ir_ui_view: action_done is not a valid action on account.payment.order

View name: account.payment.order.form
Error context:
 view: ir.ui.view(4540,)
 xmlid: account_payment_order_form
 view.model: account.payment.order
 file: /opt/odoo/kentest14-odoo/src/odoo-platform/odoo/external-src/bank-payment/account_payment_order/views/account_payment_order.xml
```
 
Getting error with latest gitaggregate

**Review by atchuthan — CHANGES_REQUESTED** [review]

**Review by dsolanki-initos — CHANGES_REQUESTED** [review]
It would be nice if resolve conflicts for all make checks passed?

**Comment by alexis-via** [comment]
For me, this PR is not relevant any more following the merge of my PR https://github.com/OCA/bank-payment/pull/822 a few days ago. It uses fields that don't exist any more...

File: account_payment_order/models/account_payment_order.py [hunk 4]
```diff
@@ -465,6 +465,10 @@ def _prepare_move_line_offsetting_account(
         else:
             account_id = self.journal_id.payment_debit_account_id.id
 
+        if self.payment_mode_id.offsetting_account == "bank_account":
```
Comment (atchuthan) [inline account_payment_order/models/account_payment_order.py:468]: offsetting_account field is not available in https://github.com/ursais/bank-payment/blob/14.0-imp-account_payment_order/account_payment_order/models/account_payment_mode.py#L78, so we are getting error.  Please check and make changes

It was removed in this commit(https://github.com/ursais/bank-payment/commit/cbe1eed5ac2a7987a3f23db1d975b6e13abab32c#diff-bc1e10412f0a69eab3d4e969864e38c2051ae55ee7a4f6fd10d1463e8ae9e72dL78)

cc: @SodexisTeam

File: account_payment_order/models/account_payment_order.py (hunk 3 above)
Comment (rousseldenis) [reply account_payment_order/models/account_payment_order.py:469]: Agree. @max3903 Could you update accordingly?

Comment (rousseldenis) [reply account_payment_order/models/account_payment_order.py:469]: @alexis-via That's a behaviour you partly removed here: https://github.com/OCA/bank-payment/pull/822/files#diff-f55b2fce8819072c65c79eaaf78564a96202ccca3de7a8be6f99c1c6862cb325L430

Comment (rousseldenis) [reply account_payment_order/models/account_payment_order.py:469]: Did it there : https://github.com/ursais/bank-payment/pull/9

Comment (rousseldenis) [reply account_payment_order/models/account_payment_order.py:469]: Ok, I confirm this has already been solved in latest version. So, no need this change

**Pull request closed without merging**